Code/GraphMol/MolDraw2D/MolDraw2D.cpp

```diff
@@ -3776,6 +3780,27 @@ pair<string, OrientType> MolDraw2D::getAtomSymbolAndOrientation(
   return std::make_pair(symbol, orient);
 }
 
+std::string getAtomListText(const Atom &atom) {
+  PRECONDITION(atom.hasQuery(), "no query");
+  PRECONDITION(atom.getQuery()->getDescription() == "AtomOr", "bad query type");
+
+  std::string res = "";
```

Comment:
Should we consider using a `stringstream` here? This looks like will be initializing new string buffers several times along the function.

Reply:
I wasn't particularly concerned about that since this doesn't feel like a time-critical bit of code, but I was curious, so I did a little benchmark:
```
/*
compile with:
g++ -I /opt/boost_1_71_0/include -L /opt/boost_1_71_0/lib -o a.out blah.cpp \
  -lboost_timer -lboost_system

*/

#include <iostream>
#include <string>
#include <sstream>
#include <boost/timer/timer.hpp>

int main(int argc, char *argv[]) {
  std::cerr << "std::string" << std::endl;
  {
    boost::timer::auto_cpu_timer t;
    double accum = 0;
    for (auto i = 0; i < 10000000; ++i) {
      std::string str = "[";
      str += "C";
      str += ",";
      str += "N";
      str += ",";
      str += "O";
      str += "]";
      accum += str.size();
    }
    std::cerr << "    " << accum << std::endl;
  }
  std::cerr << "std::stringstream" << std::endl;
  {
    boost::timer::auto_cpu_timer t;
    double accum = 0;

    for (auto i = 0; i < 10000000; ++i) {
      std::stringstream sstr;
      sstr << "[";
      sstr << "C";
      sstr << ",";
      sstr << "N";
      sstr << ",";
      sstr << "O";
      sstr << "]";
      std::string ostr = sstr.str();
      accum += ostr.size();
    }
    std::cerr << "    " << accum << std::endl;
  }
}
```
When I run this on my linux box, the std::string approach takes 0.43s, the std::stringstream version takes 3.14s. Compiling with `-O3` changes that to 0.17s and 2.80s.
So at least with g++ in this simple case the std::string approach is significantly faster.
So at least for

Reply:
switching to `-g -O0` gives the original 0.4s, 3.2s back, so it's probably not the compiler reducing the string ops to a simple assignment.

Reply:
Ok, let's keep the strings, then. Thanks for checking.